core/src/main/java/org/infinispan/stream/impl/local/EntryStreamSupplier.java

```diff
@@ -44,7 +45,9 @@ public EntryStreamSupplier(Cache<K, V> cache, ConsistentHash hash, Supplier<Stre
       }
       if (segmentsToFilter != null && hash != null) {
          log.tracef("Applying segment filter %s", segmentsToFilter);
-         stream = stream.filter(k -> segmentsToFilter.contains(hash.getSegment(k.getKey())));
+         BitSet bitSet = new BitSet(hash.getNumSegments());
```

Comment:
To be honest the way I want this is to eventually convert this over to using a BitSet completely, but I don't know if that is feasible until we also change the hash to return BitSets instead of Set<Integer> or at least some sort of native collection type so we can avoid these unboxings all over the place.